### PR TITLE
fix(BUX-250): SPV; decoding

### PR DESCRIPTION
**Pull request by arkadiuszos4chain**
fix: 
* SPV - scripts veryfication
* SPV - satoshis check - use correct input parent
* decoding - use correct fields type

# Pull Request Checklist

- [x] 📖 I created my PR using provided  : [CODE_STANDARDS](https://github.com/bitcoin-sv/go-paymail/blob/main/.github/CODE_STANDARDS.md)
- [x 📖 I have read the short Code of Conduct: [CODE_OF_CONDUCT](https://github.com/bitcoin-sv/go-paymail/blob/main/.github/CODE_OF_CONDUCT.md)
- [x] 🏠 I tested my changes locally.
- [ ] ✅ I have provided tests for my changes.
- [x] 📝 I have used conventional commits.
- [x] 📗 I have updated any related documentation.
- [x] 💾 PR was issued based on the Github or Jira issue.

### CODE REVIEW

**Review by pawellewandowski98 — APPROVED** [review]

File: p2p_beef_tx.go [hunk 1]
```diff
@@ -123,8 +123,8 @@ func decodeBUMPPathsFromStream(hexBytes []byte) ([][]BUMPLeaf, []byte, error) {
 		return nil, nil, errors.New("cannot decode BUMP paths from stream - no bytes provided")
 	}
 
-	treeHeight, bytesUsed := bt.NewVarIntFromBytes(hexBytes)
-	hexBytes = hexBytes[bytesUsed:]
+	treeHeight := hexBytes[0]
```
Comment (dorzepowski) [inline p2p_beef_tx.go:126]: Why did you changed treeHeight from VarInt to bytes?

AFAIU max number that can be stored in 1 byte is 255.
Tree of height 255 can contain 65025 transactions.
And we're aiming on 1 mln of transactions and more.

**Review by dorzepowski — CHANGES_REQUESTED** [review]

File: p2p_beef_tx.go (hunk 1 above)
Comment (pawellewandowski98) [reply p2p_beef_tx.go:126]: I think because in documentation treeHeight size equals 1 byte
https://bsv.brc.dev/transactions/0074#global

Comment (arkadiuszos4chain) [reply p2p_beef_tx.go:126]: image

Comment (arkadiuszos4chain) [reply p2p_beef_tx.go:126]: I believe we do not want to send 1 mln transactions in BUMP

**Review by dorzepowski — APPROVED** [review]

**Pull request merged**